### ISSUE TITLE
System.PlatformNotSupportedException on MacOS and Linux with dotnet test

### ISSUE THREAD

**Issue by pgrm** [fixie/fixie#228]
Hello, I'm getting the following exception when I run a fixie test project using `dotnet test` on macos and linux. It seems to work when I run it with `dotnet run` instead

```
An exception occurred while invoking executor 'executor://fixie.testadapter/': 

Message transmission mode is not supported on this platform.

System.PlatformNotSupportedException
   at System.IO.Pipes.NamedPipeServerStream.Create(String pipeName, PipeDirection direction, Int32 maxNumberOfServerInstances, PipeTransmissionMode transmissionMode, PipeOptions options, Int32 inBufferSize, Int32 outBufferSize, HandleInheritability inheritability)
   at System.IO.Pipes.NamedPipeServerStream..ctor(String pipeName, PipeDirection direction, Int32 maxNumberOfServerInstances, PipeTransmissionMode transmissionMode, PipeOptions options, Int32 inBufferSize, Int32 outBufferSize, HandleInheritability inheritability)
   at Fixie.TestAdapter.VsTestExecutor.RunTests(IMessageLogger log, IFrameworkHandle frameworkHandle, String assemblyPath, Action`1 sendCommand) in C:\dev\fixie\src\Fixie.TestAdapter\VsTestExecutor.cs:line 114
   at Fixie.TestAdapter.VsTestExecutor.RunTests(IEnumerable`1 sources, IRunContext runContext, IFrameworkHandle frameworkHandle) in C:\dev\fixie\src\Fixie.TestAdapter\VsTestExecutor.cs:line 42


Test Run Failed.
```

on macos I have the following version installed:

```
$ dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.1.302
 Commit:    41faccf259

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.15
 OS Platform: Darwin
 RID:         osx.10.15-x64
 Base Path:   /usr/local/share/dotnet/sdk/3.1.302/

Host (useful for support):
  Version: 3.1.6
  Commit:  3acd9b0cd1

.NET Core SDKs installed:
  3.1.302 [/usr/local/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.1.6 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.6 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

linux is tested via the docker image `mcr.microsoft.com/dotnet/core/sdk:3.1`

I didn't test it on Windows

Is this supposed to work? What am I doing wrong? Maybe I missed something in the documentation

**Comment by plioi** [comment]
Be sure to see the quick start here: https://github.com/fixie/fixie/wiki#quick-start

The command to use when running tests from the command line is `dotnet fixie`, from within the test project folder.

`dotnet test` does work on windows, but runs far slower and with far more confused output than `dotnet fixie`.

The work on Fixie v3 is likely to address this, but even after that releases I'd still advise running with `dotnet fixie` instead of `dotnet test`.

**Comment by pgrm** [comment]
@plioi thx - the problem is, I have my unit tests written in xUnit and wanted to write integration tests in fixie, because of the flexibility, that I can build my own setup pipeline, ...

now when I run `dotnet test Solution.sln` it also runs the fixie tests, and even if I set up a filter to not execute them, they seem to be at least loaded and that's enough to throw the error. So if you say in v3 it's addressed, then with v3 my filter should be working. 

(As a workaround now, I'm modifying the solution file to remove the fixie test projects)

**Comment by arundai-ilin** [comment]
> 
> ...
> Be sure to see the quick start here: https://github.com/fixie/fixie/wiki#quick-start
> 
> The command to use when running tests from the command line is `dotnet fixie`, from within the test project folder.
> ...

Hi,
Actually the **Fixie.Console** in the latest release ([2.2.1](https://github.com/fixie/fixie/releases/tag/2.2.1)) was still built for netcoreapp2.0 ([the target in the csproj](https://github.com/fixie/fixie/blob/2.2.1/src/Fixie.Console/Fixie.Console.csproj#L5)), so it won't run on a netcoreapp3.1 single-framework environment (like ```mcr.microsoft.com/dotnet/core/sdk:3.1```) without additionally installing the netcoreapp2.0 runtime or building one of the later Fixie commits (starting with 59ae91c2dcbe18173320a11dd48a3d4906d72c44)

**Comment by plioi** [comment]
@arundai-ilin FYI, that limitation was thankfully addressed in 2.2.2

@pgrm Fixie v3 has been released, and it no longer makes any use of the problematic "Message" named pipe mode, so I'm closing this issue as resolved in v3. Thanks for alerting me to the issue, it had a meaningful impact on the design of v3.
